### PR TITLE
Allow amp-form to update amp-state

### PR DESCRIPTION
**Pull request by choumx**
Partial for #6199.

- Allows JSON results of a form submission to be passed to `<amp-state>` 
- To support "smart buttons" use case for Bind: #3459

### CODE REVIEW

File: extensions/amp-bind/0.1/amp-state.js [hunk 1]
```diff
@@ -38,39 +38,35 @@ export class AmpState extends AMP.BaseElement {
...
-    let json;
     const children = this.element.children;
     if (children.length == 1) {
       const child = children[0];
       if (isJsonScriptTag(child)) {
-        json = tryParseJson(children[0].textContent, e => {
+        const json = tryParseJson(children[0].textContent, e => {
           user().error(TAG, 'Failed to parse state. Is it valid JSON?', e);
         });
+        this.updateState_(json, true);
```
Comment (mkhatib) [inline extensions/amp-bind/0.1/amp-state.js]: Add a comment to indicate what's the second `true` parameter is.  Like: `this.updateState_(json, /*opt_isInit*/ true);`

File: extensions/amp-form/0.1/amp-form.js [hunk 2]
```diff
@@ -251,13 +251,14 @@ export class AmpForm {
           credentials: 'include',
           requireAmpResponseSourceOrigin: true,
         }).then(response => {
-          this.actions_.trigger(this.form_, 'submit-success', null);
+          this.triggerAction_(/** success */ true, response);
```
Comment (mkhatib) [inline extensions/amp-form/0.1/amp-form.js]: nit: `/* success */` (no need for second asterisk) - same below

Comment (choumx) [inline extensions/amp-form/0.1/amp-form.js]: Done.

File: extensions/amp-bind/0.1/amp-state.js (hunk 1 above)
Comment (choumx) [inline extensions/amp-bind/0.1/amp-state.js]: Done.

**Pull request merged**